vendors/espressif/boards/esp32/ports/ble/nimble/iot_ble_hal_gatt_server.c

```diff
@@ -373,10 +381,13 @@ static int prvGATTCharAccessCb( uint16_t conn_handle,
             trans_id = ( uint32_t ) ctxt;
             if( xGattServerCb.pxRequestWriteCb != NULL )
             {
+                xSemLock = 1;
+
                 if( ( ctxt->op == BLE_GATT_ACCESS_OP_WRITE_CHR ) && ( ctxt->chr->flags & BLE_GATT_CHR_F_WRITE_NO_RSP ) )
                 {
                     need_rsp = 0;
                     trans_id = 0;
+                    xSemLock = 0;
```

Comment:
Why is this set to 0 here?

Reply:
Ohh, so basically you set `xSemLock` for all cases and reset it if the request is write without response, so this will fine. But, I think this can be done the other way for better clarity.